src/main/java/org/opensearch/neuralsearch/processor/factory/RerankProcessorFactory.java

```diff
@@ -0,0 +1,153 @@
+                RerankType attempt = RerankType.from(key);
+                return attempt;
+            } catch (IllegalArgumentException e) {
+                // Assume it's just a different field in the config, so don't do anything.
+                // If we get to the end and there were no valid RerankTypes, then we can panic.
+            }
+        }
+        throw new IllegalArgumentException("no rerank type found");
+    }
+
+    /**
+     * Factory class for context fetchers. Constructs a list of context fetchers
+     * specified in the pipeline config (and maybe the query context fetcher)
+     */
+    protected static class ContextFetcherFactory {
+
+        /**
+         * Map rerank types to whether they should include the query context source fetcher
+         * @param type the constructing RerankType
+         * @return does this RerankType depend on the QueryContextSourceFetcher?
+         */
+        public static boolean shouldIncludeQueryContextFetcher(RerankType type) {
+            switch (type) {
+                case TEXT_SIMILARITY:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /**
+         * Create necessary queryContextFetchers for this processor
+         * @param config processor config object. Look for "context" field to find fetchers
+         * @param includeQueryContextFetcher should I include the queryContextFetcher?
+         * @return list of contextFetchers for the processor to use
+         */
+        public static List<ContextSourceFetcher> createFetchers(Map<String, Object> config, boolean includeQueryContextFetcher) {
+            List<ContextSourceFetcher> fetchers = new ArrayList<>();
+            @SuppressWarnings("unchecked")
+            Map<String, Object> contextConfig = (Map<String, Object>) config.remove(CONTEXT_CONFIG_FIELD);
+            if (contextConfig == null) {
+                throw new IllegalArgumentException(String.format(Locale.ROOT, "%s field must be provided", CONTEXT_CONFIG_FIELD));
+            }
+            for (String key : contextConfig.keySet()) {
+                switch (key) {
+                    case DocumentContextSourceFetcher.NAME:
+                        Object cfg = contextConfig.get(key);
+                        if (!(cfg instanceof List<?>)) {
+                            throw new IllegalArgumentException(String.format(Locale.ROOT, "%s must be a list of strings", key));
+                        }
+                        List<?> fields = (List<?>) contextConfig.get(key);
+                        if (fields.size() == 0) {
+                            throw new IllegalArgumentException(String.format(Locale.ROOT, "%s must be nonempty", key));
+                        }
+                        List<String> strfields = fields.stream().map(field -> (String) field).collect(Collectors.toList());
+                        fetchers.add(new DocumentContextSourceFetcher(strfields));
```

Comment:
can we abstract all this in the `DocumentContextSourceFetcher.create` function? Also make it more readable with the names on local variables around what we are trying to fetch.

One better way can be add the documentation as Java doc on the create function.

Reply:
ack. did my best